### ISSUE TITLE
trailing-comma for single line regression between 3.3.0 and 3.5.0?

### ISSUE THREAD

**Issue by tomduncalf** [palantir/tslint#999]
Since upgrading to 3.5.0, I'm getting errors reported for single lines missing a trailing comma, even though the rules are set up like:

```
"trailing-comma": [true, {
  "singleline": "never",
  "multiline": "always"
}],
```

Here is an example line of code which now fails lint under 3.5.0, which used to pass in 3.3.0 with the same rules:

```
return expectBothTypesEqual(form, {
  'recurringAmount': 'recurringAmount:typeError'
})
```

- it now requires a comma after the second line, which seems incorrect.

Is this a regression or a change in the spec of this rule?

**Comment by adidahiya** [comment]
This is a change in the spec of the rule, or rather a correction of it. The intent of the trailing-comma rule is to make code diffs more precise. Multiline literals should always have trailing commas because the addition of a member should only produce a one-line diff. In your example, a 2-line diff would be produced if you added a field, like this:

```ts
return expectBothTypesEqual(form, {
  'recurringAmount': 'recurringAmount:typeError',
  'somethingElse': 'foo'
})
```

Full details here https://github.com/palantir/tslint/issues/856

**Comment by tomduncalf** [comment]
Got ya, that makes perfect sense. I thought it was because single comma
objects should never contain more than one item, so you wouldn't need to be
there as you'd never append another item. The diff thing is actually more
useful :) Will update my code, thanks

**Comment by JKillian** [comment]
It's also a lot easier when you're trying to reorder

**Comment by tomduncalf** [comment]
Indeed, sorry should have closed that myself!
